youtube_dl/extractor/rai.py

```diff
@@ -500,3 +502,108 @@ def _real_extract(self, url):
         info.update(relinker_info)
 
         return info
+
+
+class HTMLListAttrsParser(compat_HTMLParser):
+    def __init__(self):
+        compat_HTMLParser.__init__(self)
+        self.items = []
+        self._level = 0
+
+    def handle_starttag(self, tag, attrs):
+        if tag == 'li' and self._level == 0:
+            self.items.append(dict(attrs))
+        self._level += 1
+
+    def handle_endtag(self, tag):
+        self._level -= 1
+
+
+class RaiPlayRadioBaseIE(InfoExtractor):
+    _BASE = 'https://www.raiplayradio.it'
+
+    def parse_list(self, webpage):
+        parser = HTMLListAttrsParser()
+        parser.feed(webpage)
+        parser.close()
+        return parser.items
+
+    def get_playlist_iter(self, url):
+        webpage = self._download_webpage(url, url)
+        for attrs in self.parse_list(webpage):
+            title = attrs['data-title'].strip()
+            webpage = urljoin(url, attrs['data-mediapolis'])
```

Comment:
If either of these attrs is missing whole playlist extraction is broken.

Reply:
The playlist webpage is very minimal (not even a complete HTML page) and the URL appears only in those attributes. This is how the website seems to work and it seems to be reliable.
I did not figured any other way to do that.

Reply:
The extraction has been working as expected for years and the URL does not appear anywhere else, so a failure/error seems reasonable to me. Should I raise a custom error?